frontend/http/parser.go

```diff
@@ -7,28 +7,43 @@ import (
 	"github.com/chihaya/chihaya/bittorrent"
 )
 
-// ParseAnnounce parses an bittorrent.AnnounceRequest from an http.Request.
+// ParseOptions is the configuration used to parse an Announce Request.
 //
-// If allowIPSpoofing is true, IPs provided via params will be used.
-// If realIPHeader is not empty string, the first value of the HTTP Header with
+// If AllowIPSpoofing is true, IPs provided via BitTorrent params will be used.
+// If RealIPHeader is not empty string, the value of the first HTTP Header with
 // that name will be used.
-func ParseAnnounce(r *http.Request, realIPHeader string, allowIPSpoofing bool) (*bittorrent.AnnounceRequest, error) {
+type ParseOptions struct {
+	AllowIPSpoofing             bool   `yaml:"allowIPSpoofing"`
+	RealIPHeader                string `yaml:"realIPHeader"`
+	bittorrent.RequestSanitizer `yaml:",inline"`
```

Comment:
Uh, this looks a bit weird, like the actual sanitizer is embedded in this options struct, which is eventually gonna be a file.

Reply:
The Sanitizer is embedded in the ParseOptions struct. I'm not sure what you're saying when you say "which is eventually going to be a file". Can you rephrase your thought?

Reply:
Because it's serialized and deserialized to a file, and it looks like we have this active-ey Sanitizer thing in it. Doesn't look like options, if that makes sense?